### PR TITLE
Coverage 99%

### PR DESCRIPTION
**Pull request by kerycdiaz**
Increase test coverage to over 99% #26

### CODE REVIEW

**Comment by matin** [comment]
@kerycdiaz can you rebase master?

**Comment by kerycdiaz** [comment]
@matin This pull request already contains the mix with master, that is, it already has the changes of glen.

File: stpmex/base.py [hunk 1]
```diff
@@ -38,22 +38,6 @@ def _validate(field, field_value, validation, validation_value):
        field_value and validation_value < len(str(field_value)):
         return 'Length of field {} must be lower than {}'\
             .format(field, len(str(field_value)))
-    if validation == 'formatPrice' and validation_value and field_value:
```
Comment (matin) [inline stpmex/base.py]: why are you deleting this? how do you know `speid` isn't using it?

File: stpmex/__init__.py [hunk 2]
```diff
@@ -35,13 +33,9 @@ def configure(empresa: str, priv_key: str, priv_key_passphrase: str,
...
     base.WSDL_PATH = wsdl_path
 
+    base.ACTUALIZA_CLIENT = Client(wsdl_path)
     if proxy is not None:
-        session = Session()
-        session.proxies = {
-            'https': f"https://{proxy_user}:{proxy_password}@{proxy}",
-            'http':  f"http://{proxy_user}:{proxy_password}@{proxy}"
+        base.ACTUALIZA_CLIENT.transport.session.proxies = {
+            'http': "http://{proxy_user}:{proxy_password}@{proxy}",
```
Comment (matin) [inline stpmex/__init__.py]: you're missing the `f` for the f-string. We also always use single-quotes—unless the string itself has a word like "don't" that has a single-quote in the word.

File: stpmex/__init__.py [hunk 3]
```diff
@@ -35,13 +33,9 @@ def configure(empresa: str, priv_key: str, priv_key_passphrase: str,
...
+            'https':  "https://{proxy_user}:{proxy_password}@{proxy}"
         }
-        base.ACTUALIZA_CLIENT = Client(wsdl_path,
-                                       transport=Transport(session=session))
```
Comment (matin) [inline stpmex/__init__.py]: why are you removing this?

File: stpmex/base.py [hunk 4]
```diff
@@ -78,41 +62,17 @@ def __init__(self, **kwargs):
         self.empresa = STP_EMPRESA
         self.firma = None
 
-    def __dir__(self):
```
Comment (matin) [inline stpmex/base.py]: this are user easy to test. here's an example:
https://github.com/cuenca-mx/botmaker-python/blob/master/tests/test_template_messages.py#L12-L14

File: stpmex/base.py [hunk 5]
```diff
@@ -78,41 +62,17 @@ def __init__(self, **kwargs):
...
     def __getattr__(self, item):
         if item.startswith('_'):
-            return self.__getattribute__(item)
+            return self.__getattr__(item)
         return getattr(self.__object__, item)
 
-    def __ne__(self, other):
-        return not self == other
-
-    def __repr__(self):
```
Comment (matin) [inline stpmex/base.py]: example of how to test: https://github.com/cuenca-mx/botmaker-python/blob/master/tests/test_template_messages.py#L12-L14

File: stpmex/base.py [hunk 6]
```diff
@@ -78,41 +62,17 @@ def __init__(self, **kwargs):
         self.empresa = STP_EMPRESA
         self.firma = None
 
-    def __dir__(self):
-        return dir(super(Resource, self)) + dir(self.__object__)
-
-    def __dict__(self):
```
Comment (matin) [inline stpmex/base.py]: I'm 99% sure this'd break code in `speid`

File: stpmex/base.py (hunk 1 above)
Comment (kerycdiaz) [reply stpmex/base.py]: The coverage does not increase if this code is not removed, I saw that it is not used anywhere in stpmex and I thought it was not used.

I did not know that using them in tests would improve the coverage.

File: stpmex/__init__.py (hunk 3 above)
Comment (kerycdiaz) [reply stpmex/__init__.py]: The syntax of the code was until the coverage drastically dropped, I just found another way to write it.

File: stpmex/base.py (hunk 6 above)
Comment (kerycdiaz) [reply stpmex/base.py]: The coverage does not increase if this code is not removed, I saw that it is not used anywhere in stpmex and I thought it was not used.

I did not know that using them in tests would improve the coverage.

**Comment by matin** [comment]
@kerycdiaz reassigned review to @ricardo8990

**Comment by kerycdiaz** [comment]
@matin Posiblemente cierre este PR y cree otro, ya que este se descontrolo jejejeje.

**Comment by matin** [comment]
@kerycdiaz como prefieres

**Comment by ricardo8990** [comment]
@kerycdiaz por favor, no elimines código. Al contrario, agrega tests para incrementar el coverage, puedes entrar a coveralls.io para ver que líneas de código les hace falta test

**Comment by kerycdiaz** [comment]
@ricardo8990 Si fue falta de comprensión pense que eran metodos no usados, ya entendi que toca es crearles test.

**Pull request closed without merging**